components/sensor/sht3xd/sht3xd.go

```diff
@@ -0,0 +1,182 @@
+	var deps []string
+	if len(config.Board) == 0 {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "board")
+	}
+	deps = append(deps, config.Board)
+	if len(config.I2CBus) == 0 {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "i2c bus")
+	}
+	return deps, nil
+}
+
+func init() {
+	registry.RegisterComponent(
+		sensor.Subtype,
+		modelname,
+		registry.Component{Constructor: func(
+			ctx context.Context,
+			deps registry.Dependencies,
+			config config.Component,
+			logger golog.Logger,
+		) (interface{}, error) {
+			attr, ok := config.ConvertedAttributes.(*AttrConfig)
+			if !ok {
+				return nil, rdkutils.NewUnexpectedTypeError(AttrConfig{}, config.ConvertedAttributes)
+			}
+			return newSensor(ctx, deps, config.Name, attr, logger)
+		}})
+
+	config.RegisterComponentAttributeMapConverter(sensor.Subtype, modelname,
+		func(attributes config.AttributeMap) (interface{}, error) {
+			var conf AttrConfig
+			return config.TransformAttributeMapToStruct(&conf, attributes)
+		}, &AttrConfig{})
+}
+
+func newSensor(
+	ctx context.Context,
+	deps registry.Dependencies,
+	name string,
+	attr *AttrConfig,
+	logger golog.Logger,
+) (sensor.Sensor, error) {
+	b, err := board.FromDependencies(deps, attr.Board)
+	if err != nil {
+		return nil, fmt.Errorf("sht3xd init: failed to find board: %w", err)
+	}
+	localB, ok := b.(board.LocalBoard)
+	if !ok {
+		return nil, fmt.Errorf("board %s is not local", attr.Board)
+	}
+	i2cbus, ok := localB.I2CByName(attr.I2CBus)
+	if !ok {
+		return nil, fmt.Errorf("sht3xd init: failed to find i2c bus %s", attr.I2CBus)
+	}
+	addr := attr.I2cAddr
+	if addr == 0 {
+		addr = defaultI2Caddr
+		logger.Warn("using i2c address : 0x44")
+	}
+
+	s := &sht3xd{
+		name:   name,
+		logger: logger,
+		bus:    i2cbus,
+		addr:   byte(addr),
+	}
+
+	err = s.reset(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return s, nil
+}
+
+// sht3xd is a i2c sensor device that reports temperature and humidity.
+type sht3xd struct {
+	generic.Unimplemented
+	logger golog.Logger
+
+	bus  board.I2C
+	addr byte
+	name string
+}
+
+// Readings returns a list containing single item (current temperature).
+func (s *sht3xd) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
+	tryRead := func() ([]byte, error) {
+		handle, err := s.bus.OpenHandle(s.addr)
+		if err != nil {
+			s.logger.Errorf("can't open sht3xd i2c %s", err)
+			return nil, err
+		}
+		err = handle.Write(ctx, []byte{sht3xdCOMMANDPOLLINGH1, sht3xdCOMMANDPOLLINGH2})
+		if err != nil {
+			s.logger.Debug("Failed to request temperature")
+			return nil, multierr.Append(err, handle.Close())
+		}
+		buffer, err := handle.Read(ctx, 2)
+		if err != nil {
+			return nil, multierr.Append(err, handle.Close())
+		}
+		return buffer, handle.Close()
+	}
+	buffer, err := tryRead()
+	if err != nil {
+		// If error, do a soft reset and try again
+		err = s.reset(ctx)
+		if err != nil {
+			return nil, err
```

Comment:
[q] confirming this would show up as an empty map on app and an error card popping up? Will this client be primarily using app to interface with us or programming their own logic?

Reply:
Current plan is that the client will be building a bidirectional remote where all of their own logic will live. We're less interested in any particular reading, more interested in aggregating this data over long periods of time.

I do not believe that our `app` interface at present is sufficient for what they want, but we may work with app in the future to implement that if it's close.